### ISSUE TITLE
phantomjs version is not working if the page contains any javascript errors.

### ISSUE THREAD

**Issue by booma** [squizlabs/HTML_CodeSniffer#71]
Hi,

I am very happy to use phantomjs version to automate our accessibility testing.  But it does not work out if the page is having any javascript errors.  How to fix this issue.

Regards
Radhakrishnan

**Comment by andrewholgate** [comment]
Firstly, thank you. We have been using the browser version of this tool for auditing and measuring our progress towards accessibility - it's a brilliant tool.

I also set it up using PhantomJS in order to automate these audits. Unfortunately almost all sites I tested gave errors via PhantomJS without providng any audit results. One of the very few that did work was http://squizlabs.github.io/HTML_CodeSniffer/

Do you know if this is due to a limitation of PhantomJS? Do you recommend other command-line tools which will allow for integration into a CI setup?

**Comment by lwright-sq** [comment]
Hi Andrew,

What kind of PhantomJS errors were you receiving, out of curiosity?

We haven't looked into other command-line tools and you would need to build a different script obviously, but you'd be looking to start with a command-line system (or possibly Node.js module I guess) that provides a headless web browser for you.

**Comment by andrewholgate** [comment]
Hi Luke,

I have tried using the current master of HTML CodeSniffer as well as the latest stable.
I am using the latest, stable PhantomJS binary for Linux:

```
$ phantomjs --version
1.9.8
```

Here are 3 examples of errors I am receiving using different URLs:
(It is not outputting the crash reports either so I am unable to report any more details)

```
$ phantomjs ~/scripts/HTML_CodeSniffer/Contrib/PhantomJS/HTMLCS_Run.js http://www.bbc.com/ WCAG2AA

Something in HTML_CodeSniffer failed to parse. Cannot run.
PhantomJS has crashed. Please read the crash reporting guide at https://github.com/ariya/phantomjs/wiki/Crash-Reporting and file a bug report at https://github.com/ariya/phantomjs/issues/new with the crash dump file attached: /tmp/49fdc567-4f52-e22d-41a98f94-40581716.dmp
Segmentation fault (core dumped)
```

```
$ phantomjs ~/scripts/HTML_CodeSniffer/Contrib/PhantomJS/HTMLCS_Run.js https://www.google.it WCAG2A
Something in HTML_CodeSniffer failed to parse. Cannot run.
```

```
phantomjs ~/scripts/HTML_CodeSniffer/Contrib/PhantomJS/HTMLCS_Run.js https://github.com/ WCAG2AAA
TypeError: 'undefined' is not a function (evaluating 'Array.prototype.forEach.call.bind(Array.prototype.forEach)')

  https://assets-cdn.github.com/assets/frameworks-fd3bd2d0c854fa5baa64e8b390de48b1eff4b59e1f38d1b1d695c4b5d835ab04.js:29
  https://assets-cdn.github.com/assets/frameworks-fd3bd2d0c854fa5baa64e8b390de48b1eff4b59e1f38d1b1d695c4b5d835ab04.js:29
ReferenceError: Can't find variable: $

  https://assets-cdn.github.com/assets/github-46628ff6533b28dfda2aeef282f8a3502316e88499a52a67ae0dd60479e3b950.js:1
  https://assets-cdn.github.com/assets/github-46628ff6533b28dfda2aeef282f8a3502316e88499a52a67ae0dd60479e3b950.js:1
Something in HTML_CodeSniffer failed to parse. Cannot run.
PhantomJS has crashed. Please read the crash reporting guide at https://github.com/ariya/phantomjs/wiki/Crash-Reporting and file a bug report at https://github.com/ariya/phantomjs/issues/new with the crash dump file attached: /tmp/1c357b43-c09b-f8e6-4952419b-144b51f8.dmp
Segmentation fault (core dumped)
```

**Comment by lwright-sq** [comment]
Hi Andrew,

Thanks for that. I'll see what I can do. What the PhantomJS script does is inject the HTMLCS standards files from the filesystem, but currently it needs to be run from the same directory as the script, otherwise it can't find them. That makes it go off and try to include the files the other (non-minified Auditor) way, and it baulks when it can't do that.

I'll definitely take a look at seeing what PhantomJS provides in terms of system vars - surely there's something there that can give me the directory of the script so you'll be able to run it from anywhere. Stay tuned.

The current workaround to that is to make sure you run the script from the same directory as the script, ie.

    cd ~/scripts/HTML_CodeSniffer/Contrib/PhantomJS
    phantomJS ./HTMLCS_Run.js http://www.google.it/ WCAG2AA

**Comment by lwright-sq** [comment]
Hi again Andrew,

Fixed in squizlabs/HTML_CodeSniffer@98a29cf

Can you please give the repo another pull and see how that works from running with your full path to the PhantomJS script when you get a chance? Thanks.

**Comment by andrewholgate** [comment]
Fantastic! The master now works perfectly with all of the previous examples I gave.

Thanks again and keep up the great work.

**Comment by lwright-sq** [comment]
Brilliant. Thanks for the report!
